PL2/PL2-core/src/test/java/nl/tudelft/pl2016gr2/core/algorithms/FilterSnipsTest.java

```diff
@@ -36,16 +51,111 @@ public void testMakeSnip() {
 
       graph.addNode(node);
     }
+  }
 
-    // Node zeroNode = new Node(0, 1, null, 0);
-    // zeroNode.addOutlink(1);
-    // graph.addNode(zeroNode);
-
-    graph.print();
-    System.out.println("-------------------");
+  /**
+   * Test filter method. The resulting graph should have 3 nodes:
+   *    - 8
+   * 1 -
+   *    - 9
+   */
+  @Test
+  public void testFilter() {
     FilterSnips filter = new FilterSnips(graph);
     OriginalGraph filteredGraph = filter.filter();
-    // filteredGraph.print();
+    
+    assertEquals(filteredGraph.getSize(), 3);
+    Node root = filteredGraph.getRoot();
+    assertEquals(root.getId(), 1);
+    assertTrue(root.getOutlinks().contains(8));
+    assertTrue(root.getOutlinks().contains(9));
+
+    assertEquals(filteredGraph.getNode(8), graph.getNode(8));
+    assertEquals(filteredGraph.getNode(9), graph.getNode(9));
```

Comment:
Avoid using too many `assert` statements in a single tests by spitting the test (test `getOutLinks` and `getNode` separately).